data/utils/assembleLsstRaw.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python2
+#
+# LSST Data Management System
+# Copyright 2014 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""Assemble a set of LSSTSim channel images into one obs_test image
+"""
+from __future__ import absolute_import, division, print_function
+import argparse
+import glob
+import os.path
+import re
+
+import numpy
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+
+OutFileName = "raw.fits"
+SizeY = 1000  # number of pixels per amplifier in X direction (Y uses all pixels)
+
+
+def openChannelImage(dirPath, x, y):
+    """Open an LSSTSim channel raw image
+    """
+    globStr = os.path.join(dirPath, "imsim_*_R22_S00_C%d%d*" % (y, x))
+    inDecoImagePathList = glob.glob(globStr)
+    if len(inDecoImagePathList) != 1:
+        raise RuntimeError("Found %s instead of 1 file" % (inDecoImagePathList,))
+    inDecoImagePath = inDecoImagePathList[0]
+    inDecoImageFileName = os.path.basename(inDecoImagePath)
+    # calib images (which are float) have names such as imsim_2_R22_S00_C00.fits
+    # raw images (which are unsigned int) have names such as imsim_890104911_R22_S00_C00_E000....
+    if not re.match(r"imsim_\d\d\d\d\d", inDecoImageFileName):
+        raise RuntimeError("Not raw data")
+
+    print("loading %s as raw unsigned integer data" % (inDecoImageFileName,))
+    return afwImage.DecoratedImageU(inDecoImagePath)
+
+
+def assembleImage(dirPath):
+    """Make one image by combining half of amplifiers C00, C01, C10, C11 of lsstSim data
+    """
+    # views and assembly operations require a masked image, not a DecoratedImage
+    inDecoImage = openChannelImage(dirPath, 0, 0)
```

Comment:
`inDecoImage` is too close to Hungarian notation for my liking, but that's my preference.